### PR TITLE
change from using adafruit_sdcard to sdcardio

### PR DESCRIPTION
**Pull request by cjsieh**
Change from adafruit_sdcard to sdcardio .

### CODE REVIEW

**Comment by cjsieh** [comment]
This is for Issue #3309 .  Note that adafruit_sdcard has issues and fails on the PyPortal and PyGamer.

**Comment by cjsieh** [comment]
pylint errors because of upgrade of pylinit to 2.6.0 .

**Comment by ladyada** [comment]
@makermelissa hey hey this needs the new exceptionizalation

**Comment by makermelissa** [comment]
Ok, should be quick. I'll submit a PR.

**Comment by makermelissa** [comment]
The pylint updates are now merged in. @cjsieh, please pull in latest changes from repo and push again.

**Comment by FoamyGuy** [comment]
I think the previous actions issue was a temporary problem. Something in the installing dependencies section failed to download the proper file. I re-ran the jobs and it did get past that issue. 

After that it did need one minor change in the docs config file to include `sdcardio` in the list of mocked imports.

**Review by FoamyGuy — APPROVED** [review]
I tested this successfully on 
`Adafruit CircuitPython 6.0.0-alpha.3 on 2020-08-28; Adafruit PyPortal with samd51j20`

Ordinarily I would think we'd want some sort of backwards compatibility since `sdcardio` is not in a stable release yet. But it wasn't working previous as noted by the issue on the core repo noted above. Maybe something could be done in adafruit_sdcard to fix it and then backward compatibility could be provided here. 

This looks good to me. Thanks for this fix @cjsieh!

I am curious if anyone else has thoughts or ideas about the backward compatibility aspect of it. If we merge this then the library will crash on `5.3.1`. It would need to be used with a newer alpha release at this time.

File: adafruit_pyportal.py [hunk 1]
```diff
@@ -63,7 +63,7 @@
 import rtc
 import supervisor
 from adafruit_io.adafruit_io import IO_HTTP, AdafruitIO_RequestError
-import adafruit_sdcard
+import sdcardio
```
Comment (tannewt) [inline adafruit_pyportal.py:66]:
```suggestion
try:
  import sdcardio
  native_sd = True
except ImportError:
  import adafruit_sdcard as sdcardio
  native_sd = False
```

File: adafruit_pyportal.py [hunk 2]
```diff
@@ -328,10 +328,10 @@ def __init__(
 
         if self._debug:
             print("Init SD Card")
-        sd_cs = DigitalInOut(board.SD_CS)
+        sd_cs = board.SD_CS
```
Comment (tannewt) [inline adafruit_pyportal.py:331]:
```suggestion
        sd_cs = board.SD_CS
        if not native_sd:
          sd_cs = DigitalInOut(sd_cs)
```

**Review by tannewt — CHANGES_REQUESTED** [review]
Until 6.0.0 is stable I think we want this to be backwards compatible because it does work for many uses on 5.3.1. I've commented what needs to be changed for backwards compatibility I think.

**Review by tannewt — APPROVED** [review]
Thank you!

**Comment by tannewt** [comment]
@FoamyGuy Mind retesting this? Thanks!

**Comment by FoamyGuy** [comment]
I retested this. Still working successfully on `6.0.0-alpha.3` and no longer crashes on import on `5.3.1`.

**Pull request merged**